src/components/ButtonMultifunction.tsx

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import {
+  Text,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  StyleSheet,
+} from "react-native";
+import colors from "../utils/colors";
+import sizes from "../utils/sizes";
+import Icon from "react-native-vector-icons/MaterialIcons";
+import { Image } from "react-native";
+
+export interface Props extends TouchableOpacityProps {
+  disabled?: boolean;
+  text: string;
+  action: string;
+}
+
+const ButtonMultifunction: React.FC<Props> = ({
+  disabled,
+  text,
+  action,
+  ...props
+}) => {
+  let iconName = "";
+
+  switch (action) {
+    case "start":
+      iconName = "play-circle-outline";
+      break;
+    case "stop":
+      iconName = "pause-circle-outline";
+      break;
+    case "cancel":
+      iconName = "delete";
+      break;
+    case "finish":
+      iconName = "save";
+      break;
+    case "restart":
+      iconName = "play-circle-outline";
+      break;
+    default:
+      break;
+  }
+
+  const buttonStyle = [
+    styles.base,
+    action === "start" && styles.start,
+    action === "stop" && styles.stop,
+    action === "cancel" && styles.cancel,
+    action === "finish" && styles.finish,
+    action === "restart" && styles.restart,
+  ];
+
+  return (
+    <TouchableOpacity
+      {...props}
+      activeOpacity={0.8}
+      disabled={disabled}
+      style={[styles.base, buttonStyle]}
+    >
+      <Text style={styles.text}>{text}</Text>
+      <Icon name={iconName} size={24} color="white" style={styles.icon} />
```

Comment:
De acordo com o especificado no figma:
```suggestion
      <Icon name={iconName} size={32} color="white" style={styles.icon} />
```